### DISCUSSION TITLE
4.8.0 Improved Microsoft Login

### DISCUSSION THREAD

**Discussion by craftycodie** [craftycodie/MineOnline#336] [General]
- Replaced Microsoft Email & Password login flow with the Device Code Login
  - This removes the dependency on Java FX which should solve a lot of the bugs users have been experiencing including https://github.com/craftycodie/MineOnline/issues/327 and https://github.com/craftycodie/MineOnline/issues/321
- Removed the hardcoded Ampersand SMP server listing (R.I.P)

![image](https://user-images.githubusercontent.com/11706994/182934142-4428a168-0999-4ad7-adbe-8785a8fe633f.png)

<hr /><em>This discussion was created from the release <a href='https://github.com/craftycodie/MineOnline/releases/tag/4.8.0'>4.8.0 Improved Microsoft Login</a>.</em>

**Comment by reCbb** [comment]
Hello,

I am attempting to run the .jar file on my late 2015 macbook air (macOS) with java installed.
When I try to open the .jar file on desktop it only opens a window with a black screen, with no other noticeable change I've tried. I don't exactly understand how to use this .jar file on my macOS.
Any advice to fixing this issue to make the client work would be highly appreciated.
I got this working on my Windows computer and I really enjoyed the client!

Thank you!
